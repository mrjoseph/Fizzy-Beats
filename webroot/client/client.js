import { ApolloLink } from 'apollo-link';
import { withClientState } from 'apollo-link-state';
import ApolloClient from 'apollo-client';
import { createUploadLink } from 'apollo-upload-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { persistCache } from 'apollo-cache-persist';
import { defaults } from './graphql/defaults/defaults';
import { resolvers } from './graphql/resolvers/resolvers';


const cache = new InMemoryCache();
// persistCache({
//   cache,
//   storage: window.localStorage,
// });


const stateLink = withClientState({
  defaults,
  resolvers, 
});

const uploadLink = createUploadLink({
  uri: 'http://localhost:5000/graphql',
});

const link = ApolloLink.from([stateLink, uploadLink]);


export const client = new ApolloClient({
  link,
  cache,
});

