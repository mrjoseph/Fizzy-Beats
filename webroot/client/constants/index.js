const cdn = 'http://localhost:3003';

export default cdn;