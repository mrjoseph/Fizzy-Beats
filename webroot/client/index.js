import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from 'react-router-dom';
import { ApolloClient, createNetworkInterface } from 'react-apollo';

const networkInterface = createNetworkInterface({
  uri: 'http://localhost:5000/graphql'
});
const client = new ApolloClient({
  networkInterface: networkInterface
});

import Routes from './app';
const App = () => <BrowserRouter><Routes /></BrowserRouter>;
ReactDOM.render(
  <ApolloProvider client={client}>
    <App />
  </ApolloProvider>
  , document.getElementById("root"));