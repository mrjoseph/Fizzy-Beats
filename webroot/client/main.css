body {
    font-family: Arial, Helvetica, sans-serif;
    font-size:100%;
    background-color: #f1f1f1;
}
