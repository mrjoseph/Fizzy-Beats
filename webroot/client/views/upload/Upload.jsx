import React, { Component } from 'react';
import { graphql } from 'react-apollo';
import withAuth from '../../AuthService/withAuth';
import { Title } from '../../styledComponents/index.styles';
import { GET_USERS_QUERY } from '../../graphql/queries/queries';
import S3Uploader from '../../components/form/s3Uploader';

class Profile extends Component {
  render() {
    const { data, user } = this.props;
    if (data.loading) return 'Loading...';
    if (data.error) return 'Error...';
  
    const { id: userId } = user;
    return (
      <div className="container">
        <S3Uploader userId={userId} />
 
      </div>
    );
  }
}

Profile.propTypes = {};

Profile.defaultProps = {
  pathname: '',
};

export default withAuth(graphql(GET_USERS_QUERY, {
  options: props => ({
    name: 'userData',
    variables: {
      id: props.user && props.user.id,
    },
  }),
})(Profile));
