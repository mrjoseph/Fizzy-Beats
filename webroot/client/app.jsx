import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Home from './views/home/home';
import User from './views/user/user';
import About from './views/about/about';
import NotFound from './views/not-found/not-found';
import './main.css';
const App = () => (
  <Switch>
    <Route exact path="/" component={Home} />
    <Route exact path="/about" component={About} />
    <Route exact path="/user" component={User} />
    <Route component={NotFound} />
  </Switch>
);
export default App;
