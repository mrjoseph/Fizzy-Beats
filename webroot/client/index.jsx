import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import { ApolloProvider } from 'react-apollo';
import Routes from './app';
import { client } from './client';

const App = () => {
  return (
      <Router>
        <Routes />
      </Router>
  )
};

ReactDOM.render(
  <ApolloProvider client={client}>
    <App />
  </ApolloProvider>,
  document.getElementById('root'),
);

