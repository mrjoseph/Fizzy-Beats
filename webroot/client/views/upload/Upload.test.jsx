import React from 'react';
import renderer from 'react-test-renderer';
import { ApolloProvider } from 'react-apollo';
import Upload from './Upload';
import { client } from '../../client';
const props = {
  history: {
    replace: () => {},
    location: {
      search: {
        foo: 'bar',
      },
    },
  },
};

describe('<upload />', () => {
  describe('snapshot render component', () => {
    it('should console.log component', () => {
      const tree = renderer.create(
        <ApolloProvider client={client}>
         <Upload {...props} />
        </ApolloProvider>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
