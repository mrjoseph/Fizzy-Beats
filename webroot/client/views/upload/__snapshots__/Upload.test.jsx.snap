// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<upload /> snapshot render component should console.log component 1`] = `"Loading..."`;
